Modules/ArcGIS/Configuration/ArcGISConfigure.ps1

```diff
@@ -904,29 +932,42 @@ Configuration ArcGISConfigure
                     
                     $Depends += @('[ArcGIS_Service_Account]Portal_RunAs_Account')
 
-                    $HasLoadBalancer = (($AllNodes | Where-Object { $_.Role -icontains 'LoadBalancer' }  | Measure-Object).Count -gt 0)
-                    $ExternalDNSName = [System.Net.DNS]::GetHostByName($PrimaryPortalMachine).HostName
-                    if($ConfigurationData.ConfigData.Portal.SslCertifcate)
-                    {
-                        $ExternalDNSName = $ConfigurationData.ConfigData.Portal.SslCertifcate.Alias
+                    $ExternalDNSName =(Get-FQDN $PrimaryPortalMachine.NodeName) 
+                    if($Node.SslCertifcate){
+                        $ExternalDNSName = $LBMachine.SslCertifcate.Alias
```

Comment:
@shailesh91  $LBMachine is not defined here. Maybe this Needs to be ```$Node.SslCertifcate.Alias```?

Reply:
@spitzerr Yep.